File: maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur, out = nums[0], nums[0]
        for i in range(1, len(nums)):
            if cur > 0: cur = nums[i] + cur
            else: cur = nums[i]
            if cur > out: out = cur
        return out

File: maximum-subarray/maximum-subarray.scala
object Solution {
    def maxSubArray(nums: Array[Int]): Int = {
        var cur = 0;
        var out = nums(0);
        for (num <- nums) {
            if (cur > 0) cur += num;
            else cur = num;
            if (cur > out) out = cur;
        }
        out;
    }
}